src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *     parameters:
+ *       - in: path
+ *         name: postID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post.
+ *     responses:
+ *       200:
+ *         description: The number of likes for the post.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 count:
+ *                   type: integer
+ *                   description: The total number of likes.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message.
+ *                 err:
+ *                   type: string
+ *                   description: Details of the error.
+ */
 router.get('/checkLikes/:postID', countPostLikes);
+
+/**
+ * @swagger
+ * /posts/topThreeLikes/{postID}:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Retrieve the top three likes for a specific post
+ *     description: Get the first three likes for a post identified by its ID. This retrieves a subset of likes from the post.
+ *     parameters:
+ *       - in: path
+ *         name: postID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post.
+ *     responses:
+ *       200:
+ *         description: The top three likes for the post.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 likes:
+ *                   type: array
+ *                   items:
+ *                     type: object
+ *                     properties:
+ *                       _id:
+ *                         type: string
+ *                         description: The unique identifier of the like.
+ *                       user_id:
+ *                         type: string
+ *                         description: The ID of the user who liked the post.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message.
+ *                 err:
+ *                   type: string
+ *                   description: Details of the error.
+ */
 router.get('/topThreeLikes/:postID', getTopThreeLikes);
+
+/**
+ * @swagger
+ * /posts/countMyPosts:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Count the number of posts created by the current user , Must be connected as user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"